functions/function/http-request-function/src/main/java/org/springframework/cloud/fn/http/request/HttpRequestFunctionProperties.java

```diff
@@ -35,64 +34,64 @@
  */
 @Validated
 @ConfigurationProperties("http.request")
-public class HttpRequestFunctionProperties implements HttpRequestProperties {
-	private static final HttpMethod DEFAULT_HTTP_METHOD = HttpMethod.GET;
+public class HttpRequestFunctionProperties {
 
 	private static final Class<?> DEFAULT_RESPONSE_TYPE = String.class;
+
 	/**
-	 * The URL to issue an http request to, as a static value.
+	 * The type used to interpret the response.
 	 */
-	private String url;
+	private Class<?> expectedResponseType = DEFAULT_RESPONSE_TYPE;
 
 	/**
-	 * The (static) request body; if neither this nor bodyExpression is provided, the payload
-	 * will be used.
+	 * Request timeout in milliseconds.
 	 */
-	private Object body;
+	private long timeout = 30_000;
 
 	/**
-	 * The kind of http method to use.
+	 * A SpEL expression against incoming message to determine the URL to use.
 	 */
-	private HttpMethod httpMethod = DEFAULT_HTTP_METHOD;
+	private Expression urlExpression;
 
 	/**
-	 * The type used to interpret the response.
+	 * A SpEL expression to derive the request method from the incoming message.
 	 */
-	private Class<?> expectedResponseType = DEFAULT_RESPONSE_TYPE;
+	private Expression httpMethodExpression = new SpelExpressionParser().parseExpression("'GET'");
```

Comment:
We probably can just use here a `new ValueExpresison<>(HttpMethod.GET)`.
No reason to have a hard SpEL parsing against static value.